src/sitemap/handler.js

```diff
@@ -393,8 +394,84 @@ export async function sitemapAuditRunner(baseURL, context) {
   };
 }
 
+export async function convertToOpportunity(auditUrl, auditData, context) {
+  const { dataAccess, log } = context;
+
+  const opportunities = await dataAccess.Opportunity.allBySiteIdAndStatus(auditData.siteId, 'NEW');
+  let sitemapOpptyPagesWithIssues = opportunities.find((oppty) => oppty.getType() === 'sitemap-pages-with-issues');
+
+  const sitemapsWithPagesWithIssues = auditData.auditResult?.details?.issues
+    ? Object.keys(auditData.auditResult?.details?.issues)
+    : [];
+
+  const newData = [];
+  sitemapsWithPagesWithIssues.forEach((sitemapUrl) => {
+    const pagesWithIssues = auditData.auditResult.details.issues[sitemapUrl];
+    pagesWithIssues.forEach((page) => {
+      newData.push({
+        ...page,
+        sitemapUrl,
+      });
+    });
+  });
+
+  if (newData.length > 0) {
+    if (!sitemapOpptyPagesWithIssues) {
+      const opportunityData = {
+        siteId: auditData.siteId,
+        auditId: auditData.id,
+        runbook: 'https://adobe.sharepoint.com/:w:/r/sites/aemsites-engineering/Shared%20Documents/3%20-%20Experience%20Success/SpaceCat/Runbooks/Experience_Success_Studio_Sitemap_Runbook.docx?d=w6e82533ac43841949e64d73d6809dff3&csf=1&web=1&e=FTWy7t',
+        type: 'sitemap-pages-with-issues',
+        origin: 'AUTOMATON',
+        title: 'Sitemap issues found',
+        description: 'A complete and accurate sitemap helps search engines efficiently crawl and index your website pages,ensuring better visibility in search results. Fixing sitemap issues can improve the discoverability of your content.',
+        guidance: {
+          steps: [
+            'Verify each URL in the sitemap, identifying any that do not return a 200 (OK) status code.',
+            'Check RUM data to identify any sitemap pages with unresolved 3xx, 4xx or 5xx status codes – it should be none of them.',
+          ],
+        },
+        tags: ['Traffic Acquisition', 'Sitemap'],
+      };
+      try {
+        sitemapOpptyPagesWithIssues = await dataAccess.Opportunity.create(opportunityData);
+      } catch (e) {
+        log.error(`Failed to create new opportunity for siteId ${auditData.siteId} and auditId ${auditData.id}: ${e.message}`);
+        throw e;
+      }
+    } else {
+      sitemapOpptyPagesWithIssues = sitemapOpptyPagesWithIssues.setAuditId(auditData.id);
+      await sitemapOpptyPagesWithIssues.save();
+    }
+  }
+
+  if (newData.length > 0 && sitemapOpptyPagesWithIssues) {
+    const buildKey = (auditData1) => `${auditData1.siteId}|sitemap-pages-with-issues|${auditData1.auditResult.reasons.map((a) => a.value)};}`;
+
+    await syncSuggestions({
+      opportunity: sitemapOpptyPagesWithIssues,
+      newData,
+      buildKey,
+      mapNewSuggestion: (issue) => ({
+        opportunityId: sitemapOpptyPagesWithIssues.getId(),
+        type: 'CODE_CHANGE',
+        rank: issue.traffic_domain,
```

Comment:
Also checking the expected sort order with PM & UX here:
https://www.figma.com/design/lTwm2PWEC7UBTr7OMFPth7?node-id=2255-52468#1050501311

Reply:
https://wiki.corp.adobe.com/display/AEMSites/Success+Studio+Projected+Business+Impact+Metrics#SuccessStudioProjectedBusinessImpactMetrics-Invalid/MissingSitemaps

Looking at this, maybe we should consider Organic Traffic to decide the rank, since that is going to influence the impact (Projected Traffic lost)